OMCompiler/Compiler/Util/List.mo

```diff
@@ -478,7 +478,7 @@ public function appendElt<T>
   input list<T> inList;
   output list<T> outList;
 algorithm
-  outList := listAppend(inList, {inElement});
+  outList := listReverse(inElement :: listReverse(inList));
```

Comment:
@perost we think `listReverse` is linear and `listAppend` is quadratic, is that correct?

Reply:
Why would `listAppend` be quadratic? It just allocates a new list that's large enough to contain both the lists. A `listAppend` is probably a bit more expensive compared with a `listReverse`, but I'm not sure about one `listAppend` vs two `listReverse`. You'll have to benchmark that to check what's faster.

Reply:
I see. We thought `listAppend` would cons the left list element by element. But in that case I'll revert this.

Reply:
That's kind of what happens, but cons has constant complexity so cons:ing a list is still just linear. Here's the implementation of `listAppend` for reference:

https://github.com/OpenModelica/OpenModelica/blob/c94e2ebbd97fbe9b3bed444316d85702b4718338/OMCompiler/SimulationRuntime/c/meta/meta_modelica_builtin.c#L440-L464